semgrep/semgrep/core_runner.py

```diff
@@ -468,18 +473,8 @@ def dedup_output(outputs: List[RuleMatch]) -> List[RuleMatch]:
     return list({uniq_id(r): r for r in outputs}.values())
 
 
-def dedup_errors(errors: List[CoreException]) -> List[CoreException]:
-    def uniq_error_id(error: CoreException) -> Tuple[str, str, int, int, int, int]:
-        return (
-            error._check_id,
-            str(error._path),
-            error._start.line,
-            error._start.col,
-            error._end.line,
-            error._end.col,
-        )
-
-    return list({uniq_error_id(r): r for r in errors}.values())
+def dedup_errors(errors: List[SemgrepError]) -> List[SemgrepError]:
+    return list(set(errors))
```

Comment:
I assume `hash` is set up on `SemgrepError` to do the right thing here?

Reply:
Yep. Attrs handles it correctly and there are tests verifying it